src/mumble/Markdown.cpp

```diff
@@ -0,0 +1,355 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "Markdown.h"
+
+#include <QRegularExpression>
+#include <QRegularExpressionMatch>
+
+namespace Markdown {
+	// Placeholder constant
+	const QString regularLineBreakPlaceholder = QString::fromLatin1("%<\\!!linebreak!!//>@");
+
+	/// Tries to match and replace an escaped character at exactly the given offset in the string
+	///
+	/// @param str A reference to the String to work on
+	/// @param offset The offset at which the matching shall be done. This will be modified to point right after
+	/// 	replacement text, if such a replacement has been made.
+	/// @returns Whether a replacement has been made
+	bool processEscapedChar(QString &str, int &offset) {
+		static const QRegularExpression s_regex(QLatin1String("\\\\(.)"));
+
+		QRegularExpressionMatch match = s_regex.match(str, offset, QRegularExpression::NormalMatch, QRegularExpression::AnchoredMatchOption);
+
+		if (match.hasMatch()) {
+			QString replacement = QString::fromLatin1("%1").arg(match.captured(1).toHtmlEscaped());
+
+			str.replace(match.capturedStart(), match.capturedEnd() - match.capturedStart(), replacement);
+
+			offset += replacement.size();
+
+			return  true;
+		} else {
+			return false;
+		}
```

Comment:
```suggestion
		}

		return false;
```

Reply:
If you agree with the suggestion, please apply it for all instances.

Reply:
What's the advantage of doing it this way instead of in the `else`-block?

Reply:
We generally try to indent code as little as possible, to avoid ending up with a pyramid.

Not that this is the case, but I believe keeping `return false` outside the block makes it clear that it's a "catch-all".